tensorboard/webapp/metrics/data_source/metrics_data_source.ts

```diff
@@ -296,4 +313,95 @@ export class TBMetricsDataSource implements MetricsDataSource {
     const params = new URLSearchParams({tag, run, format: downloadType});
     return `/experiment/${experimentId}/data/plugin/${pluginAndRoute}?${params}`;
   }
+
+  private serializeSettings(settings: Partial<PersistableSettings>): string {
+    const serializableSettings: SerializableSettings = {
+      ignoreOutliers: settings.ignoreOutliers,
+      scalarSmoothing: settings.scalarSmoothing,
+      // TooltipSort is a string enum and has string values; no need to
+      // serialize it differently to account for their unintended changes.
+      tooltipSort: settings.tooltipSort,
+    };
+    return JSON.stringify(serializableSettings);
+  }
+
+  private deserializeSettings(
+    serialized: string
+  ): Partial<PersistableSettings> {
+    const settings: Partial<PersistableSettings> = {};
+    let unsanitizedObject: Record<string, string | number | boolean>;
+    try {
+      unsanitizedObject = JSON.parse(serialized) as Record<
+        string,
+        string | number | boolean
+      >;
+    } catch (e) {
+      return settings;
+    }
+
+    if (
+      unsanitizedObject.hasOwnProperty('scalarSmoothing') &&
+      typeof unsanitizedObject.scalarSmoothing === 'number'
+    ) {
+      settings.scalarSmoothing = unsanitizedObject.scalarSmoothing;
+    }
+
+    if (
+      unsanitizedObject.hasOwnProperty('ignoreOutliers') &&
+      typeof unsanitizedObject.ignoreOutliers === 'boolean'
+    ) {
+      settings.ignoreOutliers = unsanitizedObject.ignoreOutliers;
+    }
+
+    if (
+      unsanitizedObject.hasOwnProperty('tooltipSort') &&
+      typeof unsanitizedObject.tooltipSort === 'string'
+    ) {
+      let value: TooltipSort | null = null;
+      switch (unsanitizedObject.tooltipSort) {
+        case TooltipSort.ASCENDING:
+          value = TooltipSort.ASCENDING;
+          break;
+        case TooltipSort.DESCENDING:
+          value = TooltipSort.DESCENDING;
+          break;
+        case TooltipSort.DEFAULT:
+          value = TooltipSort.DEFAULT;
+          break;
+        case TooltipSort.NEAREST:
+          value = TooltipSort.NEAREST;
+          break;
+        default:
+        // Deliberately fallthrough; may have TooltipSort from a newer version
+        // of TensorBoard where there is an enum that this version is unaware
+        // of.
```

Comment:
When a new version of TB adds onto the TooltipSort enum, the author might not realize they need to update this function.  If keeping sanitization is desired, let's add a comment at the `TooltipSort` definition, mentioning that if it changes, then we should also change this file's function.